### PR TITLE
Added unit tests for point and line

### PR DESCRIPTION
**Pull request by vivekratnavel**
I have added unit tests for point and line. I am planning to add more tests for point and line as well other mark types. But before that wanted to check if I am going in the right direction.

### CODE REVIEW

File: test/fixtures.js [hunk 1]
```diff
@@ -60,4 +60,100 @@ f.stack.binX = {
...
+};
+
+f.points['x_and_y'] = {
+  "marktype": "point",
+  "encoding": {"x": {"name": "year","type": "O"},"y": {"name": "yield","type": "Q"}},
+  "config": {"singleHeight": 400,"singleWidth": 400,"largeBandMaxCardinality": 20},
+  "data": {"url": "data/barley.json"}
+};
+
+f.points.size = {
```
Comment (kanitw) [inline test/fixtures.js]: maybe `f.points[‘x,y,size’]`

File: test/fixtures.js [hunk 2]
```diff
@@ -60,4 +60,100 @@ f.stack.binX = {
...
+};
+
+f.points['1d_ver'] = {
+  "marktype": "point",
+  "encoding": {"y": {"name": "year","type": "O"}},
+  "config": {"singleHeight": 400,"singleWidth": 400,"largeBandMaxCardinality": 20},
+  "data": {"url": "data/barley.json"}
+};
+
+f.points['x_and_y'] = {
```
Comment (kanitw) [inline test/fixtures.js]: `‘x,y’` to make it consistent with my comment below

File: test/fixtures.js [hunk 3]
```diff
@@ -60,4 +60,100 @@ f.stack.binX = {
...
+  "encoding": {
+    "x": {"name": "year","type": "O"},
+    "y": {"name": "yield","type": "Q"},
+    "color": {"name": "yield","type": "Q"}
+  },
+  "config": {"singleHeight": 400,"singleWidth": 400,"largeBandMaxCardinality": 20},
+  "data": {"url": "data/barley.json"}
+};
+
+f.points.shape = {
```
Comment (kanitw) [inline test/fixtures.js]: `’x,y,shape’`

File: test/compiler/marks.spec.js [hunk 4]
```diff
@@ -71,5 +71,120 @@ describe('compile.marks', function() {
     });
   });
 
+  describe('point', function() {
+    describe('1D, horizontal', function() {
+      var f = fixtures.points['1d_hor'],
+          e = Encoding.fromSpec(f),
+          def = marks.point.prop(e, mockLayout, {});
+      it('should end on y axis', function() {
```
Comment (kanitw) [inline test/compiler/marks.spec.js]: 'should be centered'

File: test/compiler/marks.spec.js [hunk 5]
```diff
@@ -71,5 +71,120 @@ describe('compile.marks', function() {
...
+      it('should scale on x', function() {
+        expect(def.x).to.eql({scale: X, field: "data.year"});
+      });
+    });
+
+    describe('1D, vertical', function() {
+      var f = fixtures.points['1d_ver'],
+          e = Encoding.fromSpec(f),
+          def = marks.point.prop(e, mockLayout, {});
+      it('should end on x axis', function() {
```
Comment (kanitw) [inline test/compiler/marks.spec.js]: ‘should be centered’

File: test/fixtures.js [hunk 6]
```diff
@@ -60,4 +60,100 @@ f.stack.binX = {
...
+  "encoding": {
+    "x": {"name": "year","type": "O"},
+    "y": {"name": "yield","type": "Q"},
+    "size": {"name": "*","type": "Q","aggregate": "count"}
+  },
+  "config": {"singleHeight": 400,"singleWidth": 400,"largeBandMaxCardinality": 20},
+  "data": {"url": "data/barley.json"}
+};
+
+f.points.stroke = {
```
Comment (kanitw) [inline test/fixtures.js]: f.points[‘x,y.color’]

File: test/compiler/marks.spec.js [hunk 7]
```diff
@@ -71,5 +71,120 @@ describe('compile.marks', function() {
...
+            def = marks.point.prop(e, mockLayout, {});
+        it('should have scale for shape', function () {
+          expect(def.shape).to.eql({scale: SHAPE, field: "data.bin_yield"});
+        });
+      });
+    });
+  });
+
+  describe('line', function() {
+    describe('1D, horizontal', function() {
```
Comment (kanitw) [inline test/compiler/marks.spec.js]: I don’t think we should test 1D line as it is not a sensible encoding.  
(line should have both x and y).

**Comment by kanitw** [comment]
@vivekratnavel 

Thank you for the PR.  I’m really glad that you’re helping us improving test coverage. 

I think you’re starting with the right direction.  Also thank for sending a small and quite atomic PR, which makes reviewing very easy.  

I added a few comments and would be happy to merge this after you address those comments (which shouldn’t take too much time).   
If you find my older code, which is written in a more prototyping fashion, inconsistent with my comment please feel free to bug me to refactor them as well.  


For testing other marks, I think the code for marks should be mostly reasonable (maybe 95%) but if you find a part where the logic might not make sense, please feel free to send us questions.  
You might also find running `gulp coverage` useful for seeing how much have you help us cover :)

**Comment by vivekratnavel** [comment]
@kanitw 

Thank you for your comments. I will make required changes and let you know.

**Comment by vivekratnavel** [comment]
@kanitw 

I made all the necessary changes and added a couple of new tests for mark type 'area'. Please review the changes and let me know.

Also, when will this part of the code get executed in marks.js
```
if (e.isMeasure(X)) {
    p.x = {scale: X, field: e.field(X)};
    if (e.isDimension(Y)) {
      p.x2 = {scale: X, value: 0};
      p.orient = {value: 'horizontal'};
    }
  }
```
How to test this?

File: test/compiler/marks.spec.js (hunk 7 above)
Comment (vivekratnavel) [inline test/compiler/marks.spec.js]: Makes sense. Removed the 1D tests for line and committed.

File: test/fixtures.js [hunk 8]
```diff
@@ -60,4 +60,109 @@ f.stack.binX = {
...
 };
 
+// POINT
+
+f.points = {};
+
+f.points['1d_hor'] = {
+  "marktype": "point",
+  "encoding": {"x": {"name": "year","type": "O"}},
+  "config": {"singleHeight": 400,"singleWidth": 400,"largeBandMaxCardinality": 20},
```
Comment (kanitw) [inline test/fixtures.js]: Reading this one more time, I am guessing that we actually do not need the custom config for any of these fixtures.  

(I saw that my old code has some of these leftover as well, so I probably I need clean to them up.)

**Comment by kanitw** [comment]
Re: 
```
if (e.isMeasure(X)) {
    p.x = {scale: X, field: e.field(X)};
    if (e.isDimension(Y)) {
      p.x2 = {scale: X, value: 0};
      p.orient = {value: 'horizontal'};
    }
  }
```

To test it, you probably want to read how we define `isDimension()` (and `isMeasure()`) in `field.js`.  
and test by providing measure as x and dimension as y and vice versa.   
Since area marks with only x or y do not make sense, I will have it logs error when only x or y is provided for area mark type (#558).

To make this PR small, maybe let’s just remove “config” from those fixtures so I can merge and you can send in another PR for other tests you want to help write? :)

**Comment by vivekratnavel** [comment]
Makes sense. Removed "config" from all fixtures.

**Comment by kanitw** [comment]
Thank you so much :)

**Pull request merged**